server/provider/scheduledMiniblocks.go

```diff
@@ -132,3 +133,47 @@ func findInvalidTransactions(block *api.Block) []*transaction.ApiTransactionResu
 
 	return invalidTxs
 }
+
+// Sometimes, an invalid transaction processed in a scheduled miniblock
+// might have its smart contract result (if any) saved in the receipts unit of both blocks N and N+1.
+// This function ignores the duplicate entries in block N.
+func deduplicatePreviouslyAppearingContractResults(previousBlock *api.Block, block *api.Block) {
+	previouslyAppearingContractResultsHashes := make(map[string]struct{})
+
+	// First, gather the hashes of SCRs in "Normal" miniblocks, in block N-1.
+	for _, miniblock := range previousBlock.MiniBlocks {
```

Comment:
this function could have further been split into 2, somehow reducing the need for the comments. One assembles only relevant tx hashes based on the filtering needed and the other (that duplicates code from the first function, BTW) will determine the list of ignorable tx hashes